asistente_ladm_col/core/reports/base_report_generator.py

```diff
@@ -199,6 +204,23 @@ def validate_dependencies(self):
 
         return True
 
+    def spatial_layers_is_valid(self):
+        # It is validated that there are no geometries where its Z coordinate is null.
+        self.app.core.get_layers(self.db, self.spatial_layers_to_validate, load=False)
+
+        invalid_layers = list()
+        for layer_name in self.spatial_layers_to_validate:
+            if GeometryUtils.z_coordinate_is_nan(self.spatial_layers_to_validate[layer_name]):
+                invalid_layers.append(self.spatial_layers_to_validate[layer_name].name())
+
+        if invalid_layers:
+            self.logger.warning_msg(__name__,
+                                    QCoreApplication.translate("ReportGenerator",
+                                                               "The following layers have geometries where the z-coordinate is 'nan': {}. Please adjust in order to generate the report!".format(
```

Comment:
Users can hardly know what 'nan' means. We can make an effort to communicate with them in their language.

Reply:
the message is updated for the user, setting made in commit 604cc6e